App.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import {
+  AppRegistry,
+  Button,
+  Text,
+  View
+} from 'react-native';
+import { StackNavigator } from 'react-navigation';
+
+class AllActivitiesScreen extends Component {
```

Comment:
Extend `PureComponent` here instead of `Component`. `PureComponent` does pure-rendering. It basically overrides the `shouldComponentUpdate` lifecycle method to return `this.props.a != newProps.a` for all props; i.e. it does not call the `render` method if the shallow copy check for the props passes. This will make your app faster. So you should probably use this whenever you do not need to override the `shouldComponentUpdate` lifecycle method.'

Also https://medium.com/@esamatti/react-js-pure-render-performance-anti-pattern-fb88c101332f is a good read to avoid falling victim to common pitfalls of pure-rendering.